.github/workflows/main.yml

```diff
@@ -14,6 +14,8 @@ on:
   pull_request:
   # run pipeline on merge queue
   merge_group:
+  # run pipeline from another workflow
+  workflow_call:
```

Comment:
Why do we need to run `main.yml` in nightly build? We just need `clang-tidy` right?

Reply:
we don't need to but i'd consider it best practice.

in addition, there's no harm in enabling this option now. if you have strong feelings about it, i can remove it

Reply:
It's fine to also run main